client/api/notebook.py

```diff
@@ -21,6 +23,8 @@ def run(self, protocol, messages, **kwargs):
         return self.assignment.protocol_map[protocol].run(messages, **kwargs)
 
     def auth(self, force=False, inline=True):
+        if not force and login_with_env(self.assignment):
```

Comment:
Is the `notebook.auth` method ever used anywhere?  And yeah, I think it'd be good to try and move this to assignment.authenticate to make things simpler